tests/suites/os/tests/healthcheck/index.js

```diff
@@ -41,25 +39,38 @@ module.exports = {
       .get()
       .worker.executeCommandInContainer("rm /tmp/health", "healthcheck", ip);
 
-    await delay(1000 * 10); // may not be long enough to guarantee unhealthy status
+    // wait for 10s before checking for health status to give
+    await delay(1000 * 5);
 
-    const events = JSON.parse(
-      await this.context
-        .get()
-        .worker.executeCommandInHostOS(
-          `printf '["null"'; balena events --filter container=${state.services.healthcheck} --filter event=health_status --since 1 --until "$(date +%Y-%m-%dT%H:%M:%S.%NZ)" --format '{{json .}}' | while read LINE; do printf ",$LINE"; done; printf ']'`,
-          ip
-        )
-    );
+    let status = [];
+    // Use waitUntil, because sometimes it takes time for the container to report as unhealthy, so we want to be able to re-check
+    await this.context.get().utils.waitUntil(async () => {
```

Comment:
What's the timeout on `waitUntil`?

Reply:
@klutchell it does 20 tries, with ~30 seconds gap between each try, unless there's an outright error thrown somewhere. That makes a total of 10 minutes, which to be fair, could probably be reduced, or even better, made configurable

Reply:
in fact, the delay and the no. of attempts are actually configurable already, the numbers I gave are just the defaults